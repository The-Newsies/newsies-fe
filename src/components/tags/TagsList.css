.TagsList {
  ul {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
  }
  li {
    list-style: none;
  }
}
