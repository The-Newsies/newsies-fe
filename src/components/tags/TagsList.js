import React from 'react';
import PropTypes from 'prop-types';
import Tag from './Tag';

export default function TagsList({ tags }) {
  const tagArray = tags.map(({ name, color }) => (
    <li key={name}>
      <Tag name={name} color={color} />
    </li>
  ));

  return (
    <ul>
      {tagArray}
    </ul>
  );
}

TagsList.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired
  })).isRequired
};
