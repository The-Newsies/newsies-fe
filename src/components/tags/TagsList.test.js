import React from 'react';
import { shallow } from 'enzyme';
import TagsList from './TagsList';

describe('TagsList component', () => {
  it('renders TagsList', () => {
    const wrapper = shallow(<TagsList tags={[{ name: 'art', color: 'werwer', _id: 'erereet' }]} fetchByCategory={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
