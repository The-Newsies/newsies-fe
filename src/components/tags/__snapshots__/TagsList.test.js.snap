// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsList component renders TagsList 1`] = `
<div
  className="TagsList"
>
  <ul>
    <li
      key="erereet"
    >
      <Tag
        color="werwer"
        fetchByCategory={[Function]}
        isSelected={false}
        name="art"
        setSelectedTag={[Function]}
      />
    </li>
  </ul>
</div>
`;
